File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        q=deque()
        if not root:
            return []
        q.append(root)
        ans=[]
    
        while True:
            l =[]
            list_for_ans=[]
            while q:
                l.append(q.popleft())
                
            if not l:
                break
            for x in l:
                list_for_ans.append(x.val)
                if x.left:
                    q.append(x.left)
                if x.right:
                    q.append(x.right)
            ans.append(list_for_ans)
        return ans

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.java
class Solution {
    public String longestPalindrome(String s) {
        String ans="";
        int len = 0;

        for (int i = 0; i < s.length(); i++) {
            int left = i, right = i;
            while (left >= 0 && right < s.length() && s.charAt(left)== s.charAt(right)) {
                 if (right - left + 1 > len) {
                    len = right - left + 1;
                    ans = s.substring(left, right + 1);
                }
                left--;
                right++;
               
            }
            left=i;
            right=i+1;
            while (left >= 0 && right < s.length() && s.charAt(left)== s.charAt(right)) {
                 if (right - left + 1 > len) {
                    len = right - left + 1;
                    ans = s.substring(left, right + 1);
                }
                left--;
                right++;
               
            }
            
        }
        return ans;
    }
}
